packages/wdio-browserstack-service/src/service.ts

```diff
@@ -235,4 +236,13 @@ export default class BrowserstackService implements Services.ServiceInstance {
             log.info(`${browserString} session: ${response.body.automation_session.browser_url}`)
         })
     }
+
+    _parentSuiteName(testDescription: string, otherTestDescription: string): string {
+        const shortestLength = Math.min(testDescription.length, otherTestDescription.length)
+        let c = 0
+        while (c < shortestLength && testDescription[c] === otherTestDescription[c]) {
+            c++
+        }
+        return testDescription.substr(0, c).trim()
+    }
```

Comment:
Also since this method is not using any of the class scoped vars, I suggest to move it into a [utility method](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-browserstack-service/src/util.ts) and write a unit test for it.

Reply:
@sbley you need to merge this suggestion yourself, I can't for some reason.

Reply:
I am about to move this to the util file and add unit tests, as you suggested. I also figured out a minor flaw in the algorithm I would like to fix.

I will continue next week. Thanks.

Reply:
Awesome, thanks!

Reply:
I moved the logic to the util file and improved it so that the parent name is full words instead of only equal characters.